### ISSUE TITLE
May I ask how to understand the format of the train_set.txt?

### ISSUE THREAD

**Issue by liyang-7** [tangyuhao/GOTURN-Tensorflow#2]
Thank you for your code

Example of one line: train/target/000024.jpg, train/searching/000024.jpg, 0.29269230769230764, 0.22233115468409587, 0.7991794871794871, 0.7608061002178649

what's the meaning of the 0.29269230769230764, 0.22233115468409587, 0.7991794871794871, 0.7608061002178649?. And how did you calculate this training label form the bounding boxes of the current and previous frames?

**Comment by tangyuhao** [comment]
In your case, it means the target should be in the bounding box with top left corner ()

**Comment by tangyuhao** [comment]
Bounding box is in the form of <x1, y1, x2, y2>. In your case, it means if the second frame has size 
```(h, w) #h is height and w is width```
 the target should be in the bounding box with top left corner `(y1 * h, x1 * w)` and bottom right corner `(y2 * h, x2 * w)`

**Comment by tangyuhao** [comment]
However the network output is 10x of the original bbx. Thus it always outputs values ranging from 0 to 10. Sometimes exceeding it though.

**Comment by liyang-7** [comment]
Thank you very much! The CaffeNet resize the crop image into 227*227, the input of the network is 227*227, no matter the size(w, h) of the crop image. So the output of the network need to multiply w and h in order to regain the size of the crop image. 
Because of your code, I understand the detail of GOTURN. 
Extremely grateful!

**Comment by tangyuhao** [comment]
You are welcome~

**Comment by yihongXU** [comment]
Hi,
May I ask you why sometimes they have negative values ?
Evan in your test sest examples, I found some negative values 
test_example/target/000020.jpg,test_example/searching/000020.jpg,-0.13129449838187718,0.29418666666666665,0.5079611650485438,0.95328

thanks

**Comment by tangyuhao** [comment]
because

**Comment by tangyuhao** [comment]
because the actual object is outside the search area. I did this to test whether is have some ability to guess the actual position of an object if some part of it is not inside the searching area.

**Comment by liyang-7** [comment]
Hi, may I ask you the tensorflow version in this code?
I download the pretrained model from: GOTURN_MODEL.
Uncompress the checkpoints folder, then I got two items: checkpoint and checkpoint.ckpt-1.data-00000-of-00001
I got the error: Unsuccessful TensorSliceReader constructor: Failed to find any matching files. 
I used the tf (1.1.0)
It seems that lack of the model.ckpt.index in the checkpoints folder.
Did you used the earlier tf version?

**Comment by tangyuhao** [comment]
I used 1.0.0, but 1.1.0 should also work fine. I have added the checkpoint index file. It should fix the bug.

**Comment by mrgloom** [comment]
How `Bounding box is in the form of <x1, y1, x2, y2>, usually from 0 to 1, but exceeding this range is also fine.` bbox encoding related to encoding proposed in [paper](https://arxiv.org/pdf/1604.01802.pdf)?

Is encoding in section `4.2 Learning Motion Smoothness` is only used at train time for augmentation?

**Comment by linhandev** [comment]
so does this mean when i use the output of the network to draw a bb,  fc4/10 should be my x1,y1,x2,y2   and then use the above x1*w , y1*h , x2*w ,  y2*h as a pixel wise output

**Comment by foromer4** [comment]
Thanks for this important work!

According to the original article the results are relative to the search image  which is 
and is based on source bbox (with spacing) resized to (227,227).
B.T.W -  In the code here I do not see spacing,
so I expect the result bbox to be:
  x1 = int(arr[0] / 10 * search_bbox.width) + search_bbox.left
  y1 = int(arr[1] / 10 * search_bbox.height) + search_bbox.top
  x2 = int(arr[2] / 10 * search_bbox.width) + search_bbox.left
  y2 = int(arr[3] / 10 * search_bbox.hegiht) + search_bbox.top

And in the original code (cpp) I see indeed something like this ([Unscale](https://github.com/davheld/GOTURN/blob/ae8df641079980b6c7b05c27d0bf501becc713d4/src/helper/bounding_box.cpp#L115) and [Unceneter](https://github.com/davheld/GOTURN/blob/ae8df641079980b6c7b05c27d0bf501becc713d4/src/helper/bounding_box.cpp#L179) )

However , testing on the images seems to give wrong bboxes. Could you please clarify how to draw the result bboxes?
Thanks,
Omer
